storage/storage.go

```diff
@@ -0,0 +1,97 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"github.com/gazebo-web/gz-go/v7"
+	"github.com/pkg/errors"
+	"io"
+	"io/fs"
+	"os"
+)
+
+// Storage holds the methods to interact with a Cloud provider storage.
+type Storage interface {
+	// GetFile returns the content of file from a given path.
+	GetFile(ctx context.Context, resource Resource, path string) ([]byte, error)
```

Comment:
This should not need a `path` parameter. I believe this is a remnant of the filesystem implementation we had and is no longer necessary. Each `Storage` implementation should organize its resources as its sees fit, and it should be able to determine the path of a resource based on solely on the information contained in the `resource`.

Reply:
That's not true, `path` is used to determine which file inside the resource should be downloaded, it's just for individual file downloads.

Reply:
Ah, I see. In my mind that just be a different resource, or resource should contain a `path` or `paths` property to define the set of files it encompasses, but this is well above the scope of this.

Let's keep this as is.